src/bp/ui-studio/src/web/actions/BatchRunner.ts

```diff
@@ -29,7 +31,11 @@ const BatchRunner = (execute, { maxQueue = 100, maxInterval = 20 } = {}) => {
       })
   }
 
-  const add = resourceId => {
+  const add = (resourceId, dispatch) => {
+    if (!dispatcher) {
+      dispatcher = dispatch
```

Comment:
Why not simply pass dispatch to run directly?

Reply:
I tried to import the store directly in that file but the studio crashes. I only want to call the dispatch once for the whole batch, so i'm passing a simple reference to use later in the run method